core/src/main/java/org/elasticsearch/rest/action/cat/RestNodesAction.java

```diff
@@ -267,7 +271,15 @@ private Table buildTable(boolean fullId, RestRequest req, ClusterStateResponse s
             table.addCell(node.getVersion().toString());
             table.addCell(info == null ? null : info.getBuild().shortHash());
             table.addCell(jvmInfo == null ? null : jvmInfo.version());
+            
+            long diskUsed = fsInfo.getTotal().getTotal().getBytes() - fsInfo.getTotal().getAvailable().getBytes();
+            double diskUsedRatio = (double) diskUsed / (double) Math.max(
+                fsInfo.getTotal().getTotal().getBytes(), diskUsed); // prevent division by Zero
```

Comment:
can you explain this? when total is 0 (when would that happen?) diskUsed may be negative? what would we return here?